assets/js/components/DateRangeSelector.js

```diff
@@ -16,11 +16,16 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import { useClickAway, useKey } from 'react-use';
+
 /**
  * WordPress dependencies
  */
-import { useCallback, useEffect, useRef, useState } from '@wordpress/element';
-import { ESCAPE } from '@wordpress/keycodes';
+import { useCallback, useRef, useState } from '@wordpress/element';
+import { ESCAPE, TAB } from '@wordpress/keycodes';
```

Comment:
This is the main difference in this PR.  We listen to the `TAB` key and set the React state for the menu being open to false

See: https://github.com/google/site-kit-wp/issues/3069#issuecomment-827352447